pkg/endpoint/policy.go

```diff
@@ -724,9 +724,11 @@ func (e *Endpoint) runIPIdentitySync(endpointIP netip.Addr) {
 					return nil
 				}
 
-				IP := net.IP(endpointIP.AsSlice())
 				ID := e.SecurityIdentity.ID
-				hostIP := node.GetIPv4()
+				hostIP, ok := ip.AddrFromIP(node.GetIPv4())
+				if !ok {
+					return controller.NewExitReason(fmt.Sprintf("Failed to convert node IPv4 address %q", node.GetIPv4().String()))
```

Comment:
We can avoid the `Sprintf` here

```suggestion
					return controller.NewExitReason(`Failed to convert node IPv4 address `+ node.GetIPv4().String() + `"`)
```